src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -54,34 +54,39 @@ static void HandleNodeResolve(void * context, DnssdService * result, CHIP_ERROR
 
     DiscoveredNodeData nodeData;
 
-    Platform::CopyString(nodeData.hostName, result->mHostName);
-    Platform::CopyString(nodeData.instanceName, result->mName);
+    Platform::CopyString(nodeData.hostName, results.data()[0].mHostName);
+    Platform::CopyString(nodeData.instanceName, results.data()[0].mName);
+    nodeData.port = results.data()[0].mPort;
```

Comment:
Some comments about why each of the above are guaranteed to be unique would be best or some sort of accumulator/multiloop to check.

Minmdns knows some things because it processes a single packet, however the platform implementation seems to potentially have different everything within packets.

This means that host/instance/port/interface may all be different. 
If we choose host/instance/port to not be different (reasonable assumption, like port is unlikely to change per interface) lets document.

For in interface however, we should at least check and warn if we find different interfaces. On my machine with several network cards, the interface ID is important for LL addresses.

Reply:
API update is even better, to not allow invalid data.

I was thinking minimal is to 'detect and not allow errors' (+ log), larger change that ensures valid data is the best.

Reply:
Done by changing the API so this is a non-issue.